[Bug]: Block Explorer API reacting badly to valid AMM submissions?

### Problem encountered

I [(`d1248420772cd2fc17e43bad669bd382195b681b4f1a4c69191840b5f0950beb`)](https://explorer.fairground.wtf/parties/d1248420772cd2fc17e43bad669bd382195b681b4f1a4c69191840b5f0950beb) submitted some AMM TXs to Testnet. 

Invalid ones worked as expected - the TX appears in the Block Explorer API list transactions log, with an error code. However after tweaking my parameters I submitted a presumably valid one. Since then, Block Explorer API has never returned a newer TX.



### Observed behaviour

```shell
./vegawallet-077 transaction send --wallet test-wallet --pubkey d1248420772cd2fc17e43bad669bd382195b681b4f1a4c69191840b5f0950beb --network fairground '{  
 	"submitAmm": {
		"marketId": "fb549a66ba676b2b580accaf551e1a6ba8e4f84507a2ce8cf96a616207e43ee9",
		"commitmentAmount": "10000000",
		"proposedFee": "1",
		"slippageTolerance": "0.5",
		"concentratedLiquidityParameters": {
			"base": "700000",
			"upperBound": "735000",
			"lowerBound": "665000",
			"leverageAtUpperBound": "20",
			"leverageAtLowerBound": "20"
		}
	}      
 }'
```

This account has the relevant assets for this to be accepted, and at time of writing this market is active, and of the correct type

### Expected behaviour

- Valid AMM submissions appear in the Block Explorer Transaction list API response
- Any transactions after the valid AMM submission appear as normal

### Steps to reproduce

```bash
1. Submit a valid AMM submission that is accepted
2. Try to load new transactions from the block explorer API
```


### Software version

0.77.0-preview-7

### Log digging
`04F045CED2677F9E1EF7B2A099CA7F965F3637424165636EB6851AEE50500728` was the last INVALID tx one that I submitted and appears in the party list. Then the (presumably?) valid ones were:
- `B635C9981E4CF3C8285796CA2F9C2F1A560BFC12CA92A353626A0575D770BEBF`
- `0E5F64F4014C365BA0F534CA19F4117D5D13215B52875DD22E50C80035BC55A2`
- `024DAA7336FBDFD67210A643D26BC9F416A45B2B1DF8B96CE1A7A8528C58B04A`
- `D5AD0814733A4365849AB351138BAF446EB37EBDCDDD9FDCCF528D06A44324DD`
- `F9F1AF6FA6595D82462970A534ED845FB860116CF845F38597C735568DE49971`
- `9CC2E1F3EA839052DB1F59DB5967AEA36FA35A3BAC30D88D48FD14551EC2DB45`
- `F61E2486CB43B8875B20FD3F223AED5EE2CFA70FA012F90A45F7326CCDCB9566`
- `9964C2644947B6BB98E14EFAF7237D738B5ED9007A073C6537CE4CFC6727A630`

@wwestgarth on Slack found the log result for `B635C9981E4CF3C8285796CA2F9C2F1A560BFC12CA92A353626A0575D770BEBF`

```
panic: precommit step; +2/3 prevoted for an invalid block: wrong Block.Header.AppHash.  Expected 4FB9CCD0B5685B9547BC3F5C7DC595DE634AB4F17A4D11B0F4E90A5EE3042899, got AC3204628E652CDFD4A266C878C43D95F43014E254F9186263C3ECD1D29FD797
Jul 10 13:56:18 be visor[2651231]: goroutine 332 [running]:
Jul 10 13:56:18 be visor[2651231]: github.com/cometbft/cometbft/consensus.(*State).enterPrecommit(0xc009b6c000, 0x1d246ae, 0x0)
Jul 10 13:56:18 be visor[2651231]:         /home/runner/go/pkg/mod/github.com/cometbft/cometbft@v0.38.8/consensus/state.go:1526 +0x161f
```

--- a/core/execution/common/interfaces.go
+++ b/core/execution/common/interfaces.go
@@ -386,6 +386,8 @@ type CommonMarket interface {
 	OnMarketLiquidityV2BondPenaltyFactorUpdate(d num.Decimal)
 	OnMarketLiquidityV2ProvidersFeeCalculationTimeStep(t time.Duration)
 	OnMarketLiquidityEquityLikeShareFeeFractionUpdate(d num.Decimal)
+	OnAMMMinCommitmentQuantumUpdate(context.Context, *num.Uint)
+	OnMarketAMMMaxCalculationLevels(context.Context, *num.Uint)
 
 	// liquidity provision
 	CancelLiquidityProvision(context.Context, *types.LiquidityProvisionCancellation, string) error

--- a/core/execution/engine_netparams.go
+++ b/core/execution/engine_netparams.go
@@ -465,6 +465,10 @@ func (e *Engine) OnMarketAMMMinCommitmentQuantum(ctx context.Context, c *num.Uin
 		)
 	}
 	e.npv.ammCommitmentQuantum = c
+	for _, m := range e.allMarketsCpy {
+		m.OnAMMMinCommitmentQuantumUpdate(ctx, c.Clone())
+	}
+
 	return nil
 }
 
@@ -475,6 +479,9 @@ func (e *Engine) OnMarketAMMMaxCalculationLevels(ctx context.Context, c *num.Uin
 		)
 	}
 	e.npv.ammCalculationLevels = c
+	for _, m := range e.allMarketsCpy {
+		m.OnMarketAMMMaxCalculationLevels(ctx, c.Clone())
+	}
 	return nil
 }
 

--- a/core/execution/spot/market_callbacks.go
+++ b/core/execution/spot/market_callbacks.go
@@ -24,6 +24,10 @@ import (
 	"code.vegaprotocol.io/vega/libs/num"
 )
 
+func (m *Market) OnMarketAMMMaxCalculationLevels(ctx context.Context, c *num.Uint) {}
+
+func (m *Market) OnAMMMinCommitmentQuantumUpdate(ctx context.Context, c *num.Uint) {}
+
 func (m *Market) OnMinimalHoldingQuantumMultipleUpdate(multiplier num.Decimal) error {
 	m.minHoldingQuantumMultiplier = multiplier
 	return nil

